src/syntax/syntax.rkt

```diff
@@ -147,6 +147,7 @@
 (define-syntax-rule (define-operator name [key value] ...)
   (register-operator! 'name (list (cons 'key value) ...)))
 
+;; Slower fallback for (- x) and (- x y)
 (define (ival-sub/neg x [y #f])
   (if y (ival-sub x y) (ival-neg x)))
```

Comment:
Why do we need this? I thought we have a separate `neg` operator?

Reply:
As in I thought we desugar to that—we have abstract operators called `-` and `neg`. If not let's change it so we do.

Reply:
Hmmm... I'd have to check. That'll clean things up. I'll be able to include type information in the operator table without any issues.